RecoTracker/LSTCore/standalone/code/core/trkCore.cc

```diff
@@ -274,167 +274,256 @@ float runTrackCandidate(SDL::Event<SDL::Acc> *event, bool no_pls_dupclean, bool
-                                   bool verbose) {
-  if (hitidxs.size() != hittypes.size()) {
-    std::cout << "Error: matched_sim_trk_idxs()   hitidxs and hittypes have different lengths" << std::endl;
-    std::cout << "hitidxs.size(): " << hitidxs.size() << std::endl;
-    std::cout << "hittypes.size(): " << hittypes.size() << std::endl;
-  }
+std::vector<int> matchedSimTrkIdxs(std::vector<unsigned int> hitidxs, std::vector<unsigned int> hittypes, bool verbose, float matchfrac)
+{
+    std::vector<int> matched_idxs;
+    std::vector<float> matched_idx_fracs;
+    std::tie(matched_idxs, matched_idx_fracs) = matchedSimTrkIdxsAndFracs(hitidxs, hittypes, verbose, matchfrac);
+    return matched_idxs;
+}
 
-  std::vector<std::pair<unsigned int, unsigned int>> to_check_duplicate;
-  for (size_t i = 0; i < hitidxs.size(); ++i) {
-    auto hitidx = hitidxs[i];
-    auto hittype = hittypes[i];
-    auto item = std::make_pair(hitidx, hittype);
-    if (std::find(to_check_duplicate.begin(), to_check_duplicate.end(), item) == to_check_duplicate.end()) {
-      to_check_duplicate.push_back(item);
+//___________________________________________________________________________________________________________________________________________________________________________________________
+std::tuple<std::vector<int>, std::vector<float>> matchedSimTrkIdxsAndFracs(std::vector<unsigned int> hitidxs, std::vector<unsigned int> hittypes, bool verbose, float matchfrac)
+{
+    if (hitidxs.size() != hittypes.size())
+    {
+        std::cout << "Error: matched_sim_trk_idxs()   hitidxs and hittypes have different lengths" << std::endl;
+        std::cout << "hitidxs.size(): " << hitidxs.size() << std::endl;
+        std::cout << "hittypes.size(): " << hittypes.size() << std::endl;
     }
-  }
-
-  int nhits_input = to_check_duplicate.size();
 
-  std::vector<vector<int>> simtrk_idxs;
-  std::vector<int> unique_idxs;  // to aggregate which ones to count and test
+    std::vector<std::pair<unsigned int, unsigned int>> to_check_duplicate;
+    for (int i = 0; i < hitidxs.size(); i++)
+    {
+        auto item = std::make_pair(hitidxs[i], hittypes[i]);
+        if (std::find(to_check_duplicate.begin(), to_check_duplicate.end(), item) == to_check_duplicate.end())
+        {
+            to_check_duplicate.push_back(item);
+        }
+    }
 
-  if (verbose) {
-    std::cout << " '------------------------': "
-              << "------------------------" << std::endl;
-  }
+    int nhits_input = to_check_duplicate.size();
 
-  for (size_t ihit = 0; ihit < to_check_duplicate.size(); ++ihit) {
-    auto ihitdata = to_check_duplicate[ihit];
-    auto &&[hitidx, hittype] = ihitdata;
+    std::vector<vector<int>> simtrk_idxs;
+    std::vector<int> unique_idxs; // to aggregate which ones to count and test
 
-    if (verbose) {
-      std::cout << " hitidx: " << hitidx << " hittype: " << hittype << std::endl;
+    if (verbose)
```

Comment:
I found some places in the codes are exactly the same for before and after the changes, except the formats. Can you don't do the clang-format, so that the codes don't show a lot of git adds and deletes? To help the review process easier? Thank you!

Reply:
I've reformatted the code so that the styling is consistent, but I'm not sure if it helps with git changes. Please let me know if there's a better way to do it

Reply:
I would suggest let's merge this quick, since it's simple functions. Let's not make the formatting change and we can discuss how we want to do it in LST meetings.

Reply:
As discussed during the Hackaton: please, **do not merge this**.